src/main/java/ru/itis/services/WishListService.java

```diff
@@ -37,10 +40,12 @@ public WishList addNewWishList(String title, String token) {
     public boolean removeByTitle(String title, String token) {
         Optional<WishList> wishListCandidate = wishListRepository.findByTitle(title);
         if (wishListCandidate.isPresent()) {
-            User currentUser = userService.findUserByToken(token).get();
-            currentUser.getWishLists().remove(wishListCandidate.get());
-            userService.save(currentUser);
-            wishListRepository.delete(wishListCandidate.get());
+            if (userService.findUserByToken(token).isPresent()) {
+                User currentUser = userService.findUserByToken(token).get();
+                currentUser.getWishLists().remove(wishListCandidate.get());
+                userService.save(currentUser);
+                wishListRepository.delete(wishListCandidate.get());
+            }
             return true;
```

Comment:
Read about [command and query separation](https://en.wikipedia.org/wiki/Command%E2%80%93query_separation)

Reply:
don't agree